**Lab 1 Question 1**

Multiply the following two matrices, if possible. 
$$
   A=\left[
    \begin{array}{ccc}
      -8 & 7 & 5 \\
      -8 & 1 & 4
    \end{array}\right]\hspace{.5in}B=\left[
    \begin{array}{cc}
      1 & -8 \\
      0 & 7 \\
      0 & 1 
    \end{array}\right]
$$


In [1]:
A=matrix([[-8,7,5],[-8,1,4]])

In [2]:
B=matrix([[1,-8],[0,7],[0,1]])

In [3]:
A*B

[ -8 118]
[ -8  75]

In [4]:
B*A

[ 56  -1 -27]
[-56   7  28]
[ -8   1   4]

**Lab 1 Question 2**

Find the determinant of the following matrix.

$$
    A=\left[
    \begin{array}{ccc}
      6 & 6 & -9 \\
      -1 & -9 & -5 \\
      1 & 2 & -4 
    \end{array}\right]
$$

In [5]:
A=matrix([[6,6,-9],[-1,-9,-5],[1,2,-4]])

In [6]:
det(A)

159

**Lab 1 Question 3**

Find the determinant of the following matrix.

$$
  D=\left[
    \begin{array}{cccc}
      -5 & -4 & -5 & -6 \\
      0 & 2 & 0 & -3 \\
      -6 & 5 & 3 & 8 \\
      5 & 1 & -9 & 5 \\
    \end{array}\right]
$$

In [8]:
D=matrix([[-5,-4,-5,-6],[0,2,0,-3],[-6,5,3,8],[5,1,-9,5]])

In [9]:
det(D)

-3691

**Lab 1 Question 4**

Find the adjugate of the following matrix.

$$
  B=\left[
    \begin{array}{ccc}
      5 & -4 & -4 \\
      -1 & 0 & -7 \\
      7 & 5 & -2
    \end{array}\right]
$$

In [10]:
B=matrix([[5,-4,-4],[-1,0,-7],[7,5,-2]])

In [12]:
det(B)*~B

[ 35 -28  28]
[-51  18  39]
[ -5 -53  -4]

**Lab 1 Question 5**

Find the inverse of the following matrix.

$$
  C=\left[
    \begin{array}{ccc}
      -9 & 8 & 2 \\
      -2 & -5 & -4 \\
      7 & 5 & 5 \\
    \end{array}\right]
$$

In [13]:
C=matrix([[-9,8,2],[-2,-5,-4],[7,5,5]])

In [14]:
~C

[   5/49   30/49   22/49]
[  18/49   59/49   40/49]
[ -25/49 -101/49  -61/49]

**Lab 1 Question 6**

You receive a coded message. You know that each letter of
the original message was replaced with a one- or two-digit number
corresponding to its placement in the English alphabet, so &quot;E&quot; is
represented by &quot;5&quot; and &quot;W&quot; by &quot;23&quot;; spaces in the message are
indicated by zeroes. You also know that the message was transformed
(encoded) by multiplying the message by the following
matrix:

$$
  \left[
    \begin{array}{ccc}
    1 & 2 & 3 \\
    0 & 1 & 4 \\
    5 & 6 & 0 
  \end{array}\right]
$$

Translate the coded message:

$$
  \left[
    \begin{array}{cccccc}
    108 & 8 & 26 & 95 & 69 & 3 \\
    79 & 0 & 13 & 95 & 76 & 1 \\
    238 & 40 & 79 & 114 & 60 & 11
  \end{array}\right]
$$


In [15]:
A=matrix([[1,2,3],[0,1,4],[5,6,0]])

In [16]:
M=matrix([[108,8,26,95,69,3],[79,0,13,95,76,1],[238,40,79,114,60,11]])

In [17]:
~A*M

[20  8  5  0 12  1]
[23  0  9 19  0  1]
[14  0  1 19 19  0]

**Lab 2 Question 1**

A triangle has vertices $(1,2),(3,-4),(-2,3)$. Find the
area of the triangle.

In [18]:
T=matrix([[2,-6],[-3,1]])

In [20]:
abs((1/2)*det(T))

8

**Lab 2 Question 2**

The currents running through an electrical system are given
by the following system of equations. The three currents
$I_{1},I_{2},I_{3}$ are measured in amps. Solve the system to find the
currents in this circuit. (Use elementary row operations to find the
inverse matrix.)
$$
    \begin{array}{clclclc}
    I_{1} & + & 2I_{2} & - & I_{3} & = & 0.425 \\
    3I_{1} & - & I_{2} & + & 2I_{3} & = & 2.225 \\
    5I_{1} & + & I_{2} & + & 2I_{3} & = & 3.775
    \end{array}
$$

In [36]:
A=matrix([[1,2,-1],[3,-1,2],[5,1,2]])

In [24]:
b=matrix([[0.425],[2.225],[3.775]])

In [27]:
x=~A*b

In [29]:
x

[0.375000000000000]
[0.399999999999999]
[0.750000000000000]

$I_{1}=0.375,I_{2}=0.4,I_{3}=0.75$

**Lab 2 Question 3**

Find the equation of the parabola $y=ax^{2}+bx+c$ that
  passes through the following three points: $(-2,40),(1,7),(3,15)$.
  (Use the determinant/adjugate method to find the inverse matrix.)

In [1]:
P=matrix([[4,-2,1],[1,1,1],[9,3,1]])

In [2]:
y=matrix([[40],[7],[15]])

In [3]:
d=~P*y

In [4]:
d

[ 3]
[-8]
[12]

$y=3x^{2}-8x+12$

**Lab 2 Question 4**

The Arcadium arcade in Lynchburg, Tennessee uses 3
  different colored tokens for their game machines. For \$20 you can
  purchase any of the following mixtures of tokens: 14 gold, 20
  silver, and 24 bronze; OR, 20 gold, 15 silver, and 19 bronze; OR, 30
  gold, 5 silver, and 13 bronze. What is the monetary value of the
  bronze token? (Use Cramer's Rule.)

In [43]:
L=matrix([[14,20,24],[20,15,19],[30,5,13]])

In [44]:
d=matrix([[20,20,20]])

In [45]:
~L*d.transpose()

[ 1/2]
[7/20]
[ 1/4]

The gold token is 50 cents; the silver token is 35 cents; the bronze token is 25 cents.

**Lab 2 Question 5**

In the position function for vertical height
$$
    s(t)=\frac{1}{2}at^{2}+v_{0}t+s_{0}
$$
  $s(t)$ represents height in meters and $t$ represents time in
  seconds.

  * Find the position function for a volleyball served at an
    initial height of one meter, with height of 6.275 meters 0.5
    seconds after serve, and height of 9.1 meters one second after
    serve.
  * How long until the ball hits the ground on the other side of
    the net if everyone on that team completely misses it?


In [46]:
V=matrix([[0.125,0.5,1],[0.5,1,1],[0,0,1]])

In [47]:
h=matrix([[6.275,9.1,1]])

In [48]:
~V*h.transpose()

[-9.80000000000000]
[ 13.0000000000000]
[ 1.00000000000000]

In [50]:
t=var('t')

In [51]:
solve(-4.9*t^2+13*t+1,t)

[t == -1/49*sqrt(4715) + 65/49, t == 1/49*sqrt(4715) + 65/49]

In [52]:
n(1/49*sqrt(4715) + 65/49)

2.72787464490636

The position function is $s(t)=-4.9t^{2}+13t+1$. It will take 2.73 seconds until the ball hits the ground on the other side of the net if everyone on that team completely misses it.

**Lab 2 Question 6**

Last Tuesday, Regal Cinemas sold a total of 8500 movie
  tickets. Proceeds totaled \$64,600. Tickets can be bought in one of
  3 ways: a matinee admission costs \$5, student admission is \$6 all
  day, and regular admissions are \$8.50. How many of each type of
  ticket was sold if twice as many student tickets were sold as
  matinee tickets? (Use technology to solve the system of linear equations.)

In [53]:
R=matrix([[1,1,1],[5,6,8.5],[2,-1,0]])

In [54]:
c=matrix([[8500,64600,0]])

In [55]:
~R*c.transpose()

[899.999999999998]
[1800.00000000001]
[5800.00000000000]

The following were sold: 900 matinee tickets, 1800 student tickets, 5800 regular tickets.

**Lab 3 Question 1**

Find the equation of the quadratic
$$
    y=ax^{2}+bx+c
$$
  which passes through $(0,0),(1,2),(-1,3)$.

In [56]:
A=matrix([[0,0,1],[1,1,1],[1,-1,1]])

In [57]:
b=matrix([[0,2,3]])

In [58]:
~A*b.transpose()

[ 5/2]
[-1/2]
[   0]

The equation of the quadratic is $y=2.5x^{2}-0.5x$.

**Lab 3 Question 2**

Find the determinant for the following Hermitian matrix.
  What do you notice?
 $$
    \left[
      \begin{array}{ccc}
        4 & 3-2i & -3i \\
        3+2i & 1 & -5+2i \\
        3i & -5-2i & 2
      \end{array}\right]
 $$

In [59]:
H=matrix([[4,3-2*i,-3*i],[3+2*i,1,-5+2*i],[3*i,-5-2*i,2]])

In [60]:
H.determinant()

-239

**Lab 3 Question 3**

Solve the following systems of linear equations.
$$
    \begin{array}{ccccccc}
      3x&-&4y&+&7z&=&-23 \\
      5x&-&10y&+&11z&=&-47 \\
      5x&&&-&z&=&7 \\
      x&-&3y&+&2z&=&-12
    \end{array}
$$
<hr>
$$
    \begin{array}{ccccccc}
      2u&+&v&-&2w&=&4 \\
      2u&+&4v&-&3w&=&9 \\
      4u&+&5v&-&5w&=&-11
    \end{array}
$$

In [61]:
B=matrix([[3,-4,7,-23],[5,-10,11,-47],[5,0,-1,7],[1,-3,2,-12]])

In [62]:
B.echelon_form()

[  1   2   3   1]
[  0   5   1  13]
[  0   0  14 -28]
[  0   0   0   0]

$S=\{(x,y,z)\in\mathbb{R}^{3}|x=1,y=3,z=-2\}$

In [63]:
C=matrix([[2,1,-2,4],[2,4,-3,9],[4,5,-5,-11]])

In [64]:
C.echelon_form()

[ 2  1 -2  4]
[ 0  3 -1  5]
[ 0  0  0 24]

$S=\{\}$

**Lab 3 Question 4**

Represent the following complex numbers in polar form
  $r(\cos\theta+i\sin\theta)$.
$$
    3+4i
$$
<hr>
$$
    -1.04-1.56i
$$

In [65]:
c=3+4*i

In [66]:
abs(c)

5

In [68]:
n(arg(c)*180/pi)

53.1301023541560

$3+4i\approx{}5\cdot(\cos(53.13^{\circ})+i\cdot\sin(53.13^{\circ}))$

In [69]:
d=-1.04-1.56*i

In [70]:
abs(d)

1.87488666324127

In [72]:
n(arg(d)*180/pi+360)

236.309932474020

$-1.04-1.56i\approx{}1.8749\cdot(\cos(236.31^{\circ})+i\cdot\sin(236.31^{\circ}))$

**Lab 3 Question 5**

Solve the following problems in electrical engineering.
  
  * The impedance $Z$ (in $\Omega$) in an alternating-current circuit is given by $Z=3560/-32.4^{\circ}$. Express this in rectangular form.
  * The current in a microprocessor circuit is represented by $3.75/15.0^{\circ}\mu$A. Write this in rectangular form.
  * The voltage of a generator is represented by $2.84-1.06i$kV. Write this voltage in polar form.
  

In [83]:
A=matrix([[n(real_part(3560*(cos(-32.4*pi/180)+i*sin(-32.4*pi/180))))],[n(imag_part(3560*(cos(-32.4*pi/180)+i*sin(-32.4*pi/180))))]])

In [84]:
A

[ 3005.80741478717]
[-1907.54339012523]

$Z\approx{}3005.81-1907.54i$

In [85]:
B=matrix([[n(real_part(3.75*(cos(15*pi/180)+i*sin(15*pi/180))))],[n(imag_part(3.75*(cos(15*pi/180)+i*sin(15*pi/180))))]])

In [86]:
B

[ 3.62222184858401]
[0.970571419134453]

The current in a microprocessor circuit is approximately represented by $3.62+0.9706i$.

In [93]:
C=matrix([[abs(2.84-1.06*i)],[n(arg(2.84-1.06*i)*180/pi+360)]])

In [94]:
C

[3.03136932754820]
[339.532442400431]

The voltage of a generator is approximately $3.031\cdot(\cos(339.53^{\circ})+\sin(339.53^{\circ}))$.

**Lab 3 Question 6**

Use complex arithmetic to provide the following in
  rectangular form. 
$$
    (8i-5)(7+4i)
$$
<hr>
$$
    (\sqrt{-18}\sqrt{-4})\cdot{}3i
$$
<hr>
$$
(1+i)(1-i)^{2}
  $$
<hr>
  $$
        \frac{0.25}{3-\sqrt{-1}}
  $$
<hr>
 $$
    \frac{6+5i}{3-4i}
  $$
<hr>
  $$
    \frac{(2-i^{3})^{4}}{(i^{8}-i^{6})^{3}}+i
  $$

In [5]:
z2=(I*sqrt(18)*I*sqrt(4))*3*I

In [6]:
real_part(z2)

0

In [7]:
imag_part(z2)

-18*sqrt(2)

In [8]:
n(18*sqrt(2))

25.4558441227157

In [9]:
z6=((2-I^3)^4/(I^8-I^6)^3)+I

In [10]:
real_part(z6)

-7/8

In [11]:
imag_part(z6)

4

The answers are $-67+36i$; $-18\sqrt{2}i$; $2-2i$; $\frac{3}{40}+\frac{i}{40}$; $-\frac{2}{25}+\frac{39}{25}i$, $-\frac{7}{8}+4i$.

**Lab 4 Question 1**

Solve the following equation for $x$ and $y$.
$$
(x+y\cdot{}i)(7\cdot{}i-4)=i\cdot{}(x-5)  
$$

Expand and collect like terms for $(-4x-7y)+(6x-4y+5)i=0$.

In [95]:
A=matrix([[-4,-7],[6,-4]])

In [96]:
b=matrix([[0,-5]])

In [97]:
~A*b.transpose()

[-35/58]
[ 10/29]

$S=\{(x,y)\in\mathbb{R}^{2}|x=-\frac{35}{58},y=\frac{10}{29}\}$

**Lab 4 Question 2**

Convert the following expression to rectangular form with
and without Euler's formula,
$$
  \frac{3+\sqrt{-4}}{4-i}
$$

In [99]:
(3+2*i)/(4-i)

11/17*I + 10/17

The answer is $\frac{10}{17}+\frac{11}{17}\cdot{}i$.

**Lab 4 Question 3**

Perform the indicated operations.
$$
(1-i)^{10}  
$$
<hr>
$$
(\sqrt{3}+i)^{8}(1+i)^{5}  
$$
<hr>
$$
(\sqrt{3}-i)^{-8}  
$$

In [100]:
(1-i)^10

-32*I

In [104]:
A=matrix([[real_part((sqrt(3)+i)^8*(1+i)^5)],[imag_part((sqrt(3)+i)^8*(1+i)^5)]])

In [105]:
A

[-512*sqrt(3) + 512]
[ 512*sqrt(3) + 512]

In [106]:
B=matrix([[real_part((sqrt(3)-i)^(-8))],[imag_part((sqrt(3)-i)^(-8))]])

In [107]:
B

[        -1/512]
[-1/512*sqrt(3)]

The answers are $-32i$; $512((1-\sqrt{3})+(1+\sqrt{3})i)$; $-\frac{1}{512}\left(1+\sqrt{3}\cdot{}i\right)$. They are found by converting to polar form first, then using Euler's formula to work out the exponents, then reconverting to polar and finally to rectangular form.

Here is an alternative but equivalent solution for $(\sqrt{3}+i)^{8}(1+i)^{5}$:

<img src="la-lab-4-q3-answer.png">

**Lab 4 Question 4**

Show that $\frac{1}{2}(1+\sqrt{-3})$ is the reciprocal of
its conjugate. (The reciprocal of $x$ is $1/x$.)

$$\frac{1}{\frac{1}{2}(1-i\sqrt{3})}=\frac{2\cdot(1+i\sqrt{3})}{(1-i\sqrt{3})(1+i\sqrt{3})}=\frac{1}{2}(1+i\sqrt{3})$$

**Lab 4 Question 5**

Find the fifth roots of $\sqrt{-1024}$.

$$  \sqrt[5]{\sqrt{-1024}}=(32i)^{\frac{1}{5}}=\left(32\cdot(\cos(90^{\circ})+i\sin(90^{\circ}))\right)^{\frac{1}{5}}=\left(32\cdot{}e^{i\cdot{}90^{\circ}}\right)^{\frac{1}{5}}=2\cdot\left(\cos(18^{\circ})+i\sin(18^{\circ})\right)$$

$18^{\circ}$ is only one possible angle. The others are $90^{\circ},162^{\circ},234^{\circ}$, and $306^{\circ}$.

In [115]:
A=matrix([[n(real_part(2*(cos(18*pi/180)+i*sin(18*pi/180)))),n(imag_part(2*(cos(18*pi/180)+i*sin(18*pi/180))))],[n(real_part(2*(cos(90*pi/180)+i*sin(90*pi/180)))),n(imag_part(2*(cos(90*pi/180)+i*sin(90*pi/180))))],[n(real_part(2*(cos(162*pi/180)+i*sin(162*pi/180)))),n(imag_part(2*(cos(162*pi/180)+i*sin(162*pi/180))))],[n(real_part(2*(cos(234*pi/180)+i*sin(234*pi/180)))),n(imag_part(2*(cos(234*pi/180)+i*sin(234*pi/180))))],[n(real_part(2*(cos(306*pi/180)+i*sin(306*pi/180)))),n(imag_part(2*(cos(306*pi/180)+i*sin(306*pi/180))))]])

In [116]:
A

[ 1.90211303259031 0.618033988749895]
[0.000000000000000  2.00000000000000]
[-1.90211303259031 0.618033988749895]
[-1.17557050458495 -1.61803398874989]
[ 1.17557050458495 -1.61803398874989]

The roots are $1.90+0.62i,2i,-1.90+0.62i,-1.18-1.62i,1.18-1.62i$. You can verify that they are the vertices of a pentagon.

**Lab 4 Question 6**

Consider the vector space of quadratic polynomials
$$
V=\{f|f(x)=ax^{2}+bx+c\}  
$$
Are the following three quadratic polynomials a basis for $V$?
$$
  \begin{array}{rcl}
    f_{1}(x)&=&3x^{2}+x+6 \\
    f_{2}(x)&=&5x^{2}+x+11 \\
    f_{3}(x)&=&-2x^{2}-6x+4
  \end{array}
$$

* If yes, find the coordinates in terms of this basis for $7x^{2}-x-4$.
* If no, express one of the $f_{i}$ by the others, $i=1,2,3$. 


Consider the determinant of the following matrix.

In [117]:
G=matrix([[3,5,-2],[1,1,-6],[6,11,4]])

In [118]:
G.determinant()

0

This means that the three polynomials are linearly dependent. Set up the following augmented matrix to express $f_{3}$ as a linear combination of $f_{1}$ and $f_{2}$.

In [120]:
A=matrix([[3,5,-2],[1,1,-6],[6,11,4]])

In [121]:
A.echelon_form()

[  1   0 -14]
[  0   1   8]
[  0   0   0]

Consequently, 
$$
f_{3}=(-14)\cdot{}f_{1}+8\cdot{}f_{2}
$$

**Lab 4 Question 7**

Consider the vector space of lines in $\mathbb{R}^{3}$
going through the origin.
$$
W=\{L|L:ax+by+cz=0\}  
$$
Are the following three lines a basis for $W$?
$$
  \begin{array}{rcl}
    L_{1}&:&2x+5y+z=0 \\
    L_{2}&:&x-y-2z=0 \\
    L_{3}&:&-3x+4z=0
  \end{array}
$$

* If yes, find the coordinates in terms of this basis for $L:14x+9y-12z=0$.
* If no, express one of the $L_{i}$ by the others, $i=1,2,3$. 


Consider the determinant of the following matrix.

In [123]:
H=matrix([[2,1,-3],[5,-1,0],[1,-2,4]])

In [124]:
H.determinant()

-1

This means that the three lines are linearly independent. Set up the following system of linear equations to express $L$ as a linear combination of $L_{1},L_{2}$ and $L_{3}$.

In [126]:
v=matrix([14,9,-12])

In [129]:
~H*v.transpose()

[ 2]
[ 1]
[-3]

Consequently,
$$
L=2L_{1}+L_{2}-3L_{3}
$$

**Lab 5 All Questions**

The solutions are available in D2L. The filename is <tt>la-lab-05-answers.pdf</tt>.